sql/core/src/main/scala/org/apache/spark/sql/execution/python/UserDefinedPythonFunction.scala

```diff
@@ -55,3 +56,36 @@ case class UserDefinedPythonFunction(
     }
   }
 }
+
+/**
+ * A user-defined Python table function. This is used by the Python API.
+ */
+case class UserDefinedPythonTableFunction(
+    name: String,
+    func: PythonFunction,
+    returnType: StructType,
+    udfDeterministic: Boolean) {
+
+  def builder(e: Seq[Expression]): LogicalPlan = {
+    val udtf = PythonUDTF(
+      name = name,
+      func = func,
+      elementSchema = returnType,
+      children = e,
+      udfDeterministic = udfDeterministic)
+    Generate(
+      udtf,
+      unrequiredChildIndex = Nil,
+      outer = false,
+      qualifier = None,
+      generatorOutput = Nil,
+      child = OneRowRelation()
```

Comment:
Is the child always one row relation, or is it temporary? `BatchEvalPythonUDTFExec` implementation seems like it can take many rows.

Reply:
The child will always be one row relation, but the UDTF can be used together with LATERAL join, which will be optimized during logical planning:
```
LateralJoin 
+- LocalRelation 
+- Generate [UDTF]
    +- OneRowRelation
```
to 
```
Generate [UDTF] 
+- LocalRelation
```